lib/rex/parser/fs/ntfs.rb

```diff
@@ -0,0 +1,243 @@
+      # INDEX_ALLOCATION
+      #
+      def parse_index_list(index_record, index_allocation_attribute)
+        offset_index_entry_list = index_record[0, 4].unpack("<I")[0]
+        index_size =  index_record[offset_index_entry_list + 8, 2].unpack("S")[0]
+        index_entry = index_record[offset_index_entry_list, index_size]
+        res = {}
+        while index_entry[12, 4].unpack("<I")[0] & 2 != 2
+          res.update(parse_index(index_entry))
+          # if son
+          if index_entry[12, 4].unpack("<I")[0] & 1 == 1
+            # should be 8 bytes length
+            vcn =  index_entry[-8, 4].unpack("I")[0]
+            res_son = parse_index_list(index_allocation_attribute[vcn * @bytes_per_cluster + 24, index_size * @bytes_per_cluster], index_allocation_attribute)
+            res.update(res_son)
+          end
+          offset_index_entry_list += index_size
+          index_size =  index_record[offset_index_entry_list + 8, 2].unpack("S")[0]
+          index_entry = index_record [offset_index_entry_list, index_size]
+        end
+        # if son on the last
+        if index_entry[12, 4].unpack("<I")[0] & 1 == 1
+          # should be 8 bytes length
+          vcn =  index_entry[-8, 4].unpack("I")[0]
+          res_son = parse_index_list(index_allocation_attribute[vcn * @bytes_per_cluster + 24, index_size * @bytes_per_cluster], index_allocation_attribute)
+          res.update(res_son)
+        end
+        res
+      end
+
+      #
+      # return the list of files in attribute directory and their MFT number and size
+      #
+      def index_list_from_attributes(attributes)
+        index_root_attribute = attributes[144]
+        index_record = index_root_attribute[16, index_root_attribute.length - 16]
+        if attributes.key?(160)
+          return parse_index_list(index_record, attributes[160])
+        else
+          return parse_index_list(index_record, "")
+        end
+      end
+
+      def cluster_from_attribute_non_resident(attribute, cluster_num = 0, size_max = ((2**31) - 1))
+        lowvcn = attribute[16, 8].unpack("<Q")[0]
+        highvcn = attribute[24, 8].unpack("<Q")[0]
+        offset = attribute[32, 2].unpack("<S")[0]
+        real_size = attribute[48, 8].unpack("<Q")[0]
+        attribut = ""
+        run_list_num = lowvcn
+        old_offset = 0
+
+        while run_list_num <= highvcn
+          first_runlist_byte = attribute[offset].ord
+          run_offset_size = first_runlist_byte >> 4
+          run_length_size = first_runlist_byte & 15
+          run_length = attribute[offset + 1, run_length_size]
+          run_length += "\x00" * (8 - run_length_size)
+          run_length = run_length.unpack("<Q")[0]
+
+          offset_run_offset = offset + 1 + run_length_size
+          run_offset = attribute[offset_run_offset, run_offset_size]
+          if run_offset[-1].ord & 128 == 128
+            run_offset += "\xFF".force_encoding("BINARY") * (8 - run_offset_size)
```

Comment:
You might be able to avoid using force_encoding by using the magic comment 

`# -*- coding: binary -*-`